test/TEST-38-FREEZER/testsuite.sh

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+systemd-analyze log-level debug
+systemd-analyze log-target console
+
+unit=foo.service
+unit_file=/etc/systemd/system/${unit}
```

Comment:
This name is fixed. Creating the unit dynamically is not exactly wrong, but OTOH, I don't think we gain anything by this complication. Please just rename this to e.g. `testsuite-38-foo.service` and put it in a static file alongside `testsuite-38.service`.